actionview/lib/action_view/renderer/partial_renderer.rb

```diff
@@ -405,7 +405,11 @@ def setup(context, options, block)
     def collection_from_options
       if @options.key?(:collection)
         collection = @options[:collection]
-        collection.respond_to?(:to_ary) ? collection.to_ary : []
+        if collection.respond_to?(:to_ary)
+          collection = collection.to_ary
+        else
+          raise ArgumentError.new("'#{collection.inspect}' is not an ActiveModel-compatible object. It must implement :to_ary.")
```

Comment:
@matthewd what about change like that?
```ruby
raise ArgumentError.new("#{collection.inspect} is invalid for collection. It must respond to :to_ary")
```

Inspect output is needed because if the collection is nil and without inspect output, error message will be blank.

without inspect:
```ruby
 is invalid for collection. It must respond to :to_ary
```
with inspect:
```ruby
nil is invalid for collection. It must respond to :to_ary
```